server/src/test/java/org/elasticsearch/index/replication/RecoveryDuringReplicationTests.java

```diff
@@ -378,7 +377,7 @@ public void testResyncAfterPrimaryPromotion() throws Exception {
                     assertThat(source.source.utf8ToString(), is("{ \"f\": \"normal\"}"));
                 }
             }
-            assertThat(translogOperations, is(initialDocs + extraDocs));
```

Comment:
I spent some time to separate these two expectations, but it was more complicated than I thought. I think it's fine to assert like this.